references/detection/transforms.py

```diff
@@ -437,3 +439,67 @@ def forward(
                 )
 
         return image, target
+
+
+class SimpleCopyPaste(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def combine_masks(self, masks):
+        return masks.sum(dim=0).greater(0)
+
+    def forward(self, batch: torch.Tensor, target: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+
+        # validate inputs
+        if batch.ndim != 4:
+            raise ValueError(f"Batch ndim should be 4. Got {batch.ndim}.")
+        if not batch.is_floating_point():
+            raise TypeError(f"Batch dtype should be a float tensor. Got {batch.dtype}.")
+
+        # create copy of batch and target as the original will be modified
+        batch_rolled = batch.roll(1, 0).detach().clone()
+        target_rolled = copy.deepcopy(target[-1:] + target[:-1])
+
+        # collect binary paste masks for all images
+        paste_masks = []
+
+        for source_image, paste_image, source_data, paste_data in zip(batch, batch_rolled, target, target_rolled):
+            number_of_masks = len(paste_data["masks"])
+            random_selection = torch.randint(0, number_of_masks, (number_of_masks,)).unique()
+
+            paste_alpha_mask = self.combine_masks(paste_data["masks"][random_selection])
+            paste_masks.append(paste_alpha_mask)
+
+            # update original masks
+            source_data["masks"] = source_data["masks"] ^ paste_alpha_mask & source_data["masks"]
+
+            # remove masks where no annotations are present (all values are 0)
+            mask_filter = source_data["masks"].sum((2, 1)).not_equal(0)
+            source_data["masks"] = source_data["masks"][mask_filter]
+            source_data["boxes"] = ops.masks_to_boxes(source_data["masks"])
+            source_data["labels"] = source_data["labels"][mask_filter]
+            source_data["area"] = source_data["area"][mask_filter]
+            source_data["iscrowd"] = source_data["iscrowd"][mask_filter]
+
+            # TODO: update area
+
+            # concatenate paste data with original data
+            source_data["masks"] = torch.cat((source_data["masks"], paste_data["masks"][random_selection]))
+            source_data["boxes"] = torch.cat((source_data["boxes"], paste_data["boxes"][random_selection]))
+            source_data["labels"] = torch.cat((source_data["labels"], paste_data["labels"][random_selection]))
+            source_data["area"] = torch.cat((source_data["area"], paste_data["area"][random_selection]))
+            source_data["iscrowd"] = torch.cat((source_data["iscrowd"], paste_data["iscrowd"][random_selection]))
+
+        # update the original images with paste images
+        paste_masks = torch.stack(paste_masks).to(torch.uint8)
+        paste_masks = T.GaussianBlur((5, 5), sigma=2)(paste_masks)  # Adds Gaussian Filter
```

Comment:
Can't we use functional op here instead of class and call ?

Reply:
@vfdev-5 Thanks for pointing that out. I have replaced it with the functional API.